arviz/plots/backends/matplotlib/bpvplot.py

```diff
@@ -83,6 +88,15 @@ def plot_bpv(
         obs_vals = obs_vals.flatten()
         pp_vals = pp_vals.reshape(total_pp_samples, -1)
 
+        if obs_vals.dtype.kind == "i" or pp_vals.dtype.kind == "i":
+            x = np.linspace(0, 1, len(obs_vals))
+            csi = CubicSpline(x, obs_vals)
+            obs_vals = csi(np.linspace(0.001, 0.999, len(obs_vals)))
+
+            x = np.linspace(0, 1, len(pp_vals))
+            csi = CubicSpline(x, pp_vals)
+            pp_vals = csi(np.linspace(0.001, 0.999, len(pp_vals)))
```

Comment:
Not sure about each quantity having an independent interpolation. IIUC, the interpolation is to smooth things out and convert each integer to a set of floats which are close/close-ish to each other but are not the same, here however, in the observed data there will generally not be too many points, so the effect of the interpolation and its smoothing will be quite small, leaving wavy kdes. Here is how it looks for the rugby dataset using the `p value` kind:

![image](https://user-images.githubusercontent.com/23738400/93951420-922a3880-fd46-11ea-8b71-728cfab3588f.png)

Reply:
You will get a similar plot with continuous values too. This is for a normal model with 60 observed data points.
![dada](https://user-images.githubusercontent.com/1338958/94014031-5b254880-fd81-11ea-84f6-7cbdc4cf1e71.png)

That's is related to the KDE. If I remember correctly `fast_kde` had an ad-hoc trick to be smoother in the present of a few data points, but was completely trickery so it is not part of our new KDE.

The main problem with discrete values an p-values was that they were completely off, like in the example above where the expected value was ~0.5 but the computed one was ~0.75. interpolation fix that.

How do you suggest using a single interpolation?